### PR TITLE
Add Quilt support for Technic modpacks

### LINKED ISSUE #683: some technic modpacks don't work
### Operating System

Windows, macOS, Linux, Other

### Version of PolyMC

PolyMC 1.3.1

### Description of bug

Some technic modpacks don't work correctly (when launching after install it's just vanillla)

One of these modpacks is [Sajmon's Quilted Modpack](https://www.technicpack.net/modpack/sajmons-quilted-modpack.1923907)

### Steps to reproduce

.

### Suspected cause

Maybe no missing support for some modloaders like Quilt

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

### PR DESCRIPTION
**Pull request by kthchew**
Fixes #683 (+ a minor refactor)
CC @DioEgizio

### CODE REVIEW

**Review by Scrumplex — APPROVED** [review]

File: launcher/modplatform/technic/TechnicPackProcessor.cpp [hunk 1]
```diff
@@ -185,13 +185,22 @@ void Technic::TechnicPackProcessor::run(SettingsObjectPtr globalSettings, const
...
             }
-            else if (libraryName.startsWith("net.minecraftforge:minecraftforge:"))
+            else
             {
-                components->setComponentVersion("net.minecraftforge", libraryName.section(':', 2));
-            }
-            else if (libraryName.startsWith("net.fabricmc:fabric-loader:"))
-            {
-                components->setComponentVersion("net.fabricmc.fabric-loader", libraryName.section(':', 2));
+                static QSet<QString> possibleLoaders{
```
Comment (flowln) [inline launcher/modplatform/technic/TechnicPackProcessor.cpp]:
```suggestion
                static QStringList possibleLoaders{
```

**Review by flowln — APPROVED** [review]
just a minor nitpick, looks good otherwise :)

**Comment by DioEgizio** [comment]
yes seems to work fine

**Comment by glowiak** [comment]
why is my account flagged?

**Review by txtsd — APPROVED** [review]

**Pull request merged**